api/src/main/java/org/apache/iceberg/io/ClosingIterator.java

```diff
@@ -25,33 +25,30 @@
 
 public class ClosingIterator<T> implements Iterator<T> {
```

Comment:
@rdblue any reason this is not `Closeable`? The only way for the underlying resource to be closed right now is for this iterator to be exhausted. Should this be `Closable` so the resources can be closed w/o having to read the whole iterator? Or is this by design and we should add javadoc to document this contract?

Reply:
I think this is by design. The intent must be to avoid leaks because the developer forgot to call `close()` on the iterator.

Reply:
What about the case when the developer doesn't consume the entire iterator but has scheduled this to be closed?

Reply:
If a premature close is required, the underlying `CloseableIterator` should be used directly. I see this class as a convenience wrapper for the case when all of the elements are intended to be consumed. I've added Javadoc mentioning this.

Reply:
I think this was an oversight. I don't see a reason why it shouldn't be `Closeable`.